web/api/v1/api.go

```diff
@@ -372,6 +372,24 @@ func (api *API) query(r *http.Request) (result apiFuncResult) {
 		return apiFuncResult{nil, returnAPIError(res.Err), res.Warnings, qry.Close}
 	}
 
+	if res.Value.Type() == "vector" {
+		vec, ok := res.Value.(promql.Vector)
+		if !ok {
+			panic("Error converting to vector")
+		}
+		sort.Slice(vec, func(i, j int) bool {
+			return vec[i].V > vec[j].V
+		})
+	} else if res.Value.Type() == "matrix" {
+		mat, ok := res.Value.(promql.Matrix)
+		if !ok {
+			panic("Error converting to matrix")
+		}
+		sort.Slice(mat, func(i, j int) bool {
+			return (sum(mat[i]) > sum(mat[j]))
```

Comment:
nice! I think let's go one step further and make sure the ranking function is under some variable / interface so we can quickly switch it  (: WDYT?

Maybe I want sum, maybe avg, maybe last sample, maybe first sample (:

Reply:
Yes that makes sense! Will add that in :slightly_smiling_face: